### PR TITLE
Upgrade to 16.13.0 for node

### LINKED ISSUE #9325: Node version should move to 16 to follow VS code

### PR DESCRIPTION
**Pull request by rchiodo**
Fixes #9325 

After this goes through you'll need to:
- Delete your node_modules folder
- Install node v16.13 or greater
- Reinstall your node_modules with npm install

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2129,10 +2134,14 @@
         "requirejs": "^2.3.6",
         "rxjs": "^6.5.4",
         "rxjs-compat": "^6.5.4",
+        "safe-buffer": "^5.2.1",
         "sanitize-filename": "^1.6.3",
+        "save-dev": "^0.0.1-security",
```
Comment (DonJayamanne) [inline package.json:2139]: Just double checking whether this was required or whehter you accidentally ran `npm i save-dev` instead of `npm i --save-dev`

Comment (rchiodo) [reply package.json:2139]: oh that's probably a mistake. I'll remove it

**Review by DonJayamanne — APPROVED** [review]

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
Looks like azure pipelines are failing https://dev.azure.com/monacotools/Monaco/_build/results?buildId=161564&view=results

I ran azure piplines with publish stage turned off against this branch and looks like the install phase fails.
Here's the error `npm ERR! ajv-keywords not accessible from webpack:schema-utils`

File: package.json [hunk 2]
```diff
@@ -2088,13 +2091,15 @@
         "@nteract/messaging": "^7.0.0",
         "@vscode/jupyter-ipywidgets": "^1.0.8",
         "@vscode/jupyter-lsp-middleware": "^0.2.38",
+        "ajv-keywords": "^5.1.0",
```
[the remark this answers is not in the document]
Comment (rchiodo) [reply package.json:2094]: It looks to me like these transitive dependencies weren't pulled in automatically. Perhaps the package.json of the one's that included it didn't work correctly in npm 8.

**Review by IanMatthewHuff — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -2078,16 +2081,17 @@
     "dependencies": {
         "@enonic/fnv-plus": "^1.3.0",
         "@fluentui/react": "^7.160.1",
-        "@jupyter-widgets/base": "^4.0.0",
-        "@jupyter-widgets/controls": "^3.0.0",
-        "@jupyter-widgets/schema": "^0.4.0",
-        "@jupyterlab/coreutils": "^5.1.17",
-        "@jupyterlab/nbformat": "^3.1.17",
-        "@jupyterlab/services": "^6.1.17",
+        "@jupyter-widgets/base": "4.0.0",
```
Comment (rchiodo) [inline package.json:2084]: When regenerating the package-lock.json, I had to force these to a specific version (or it picks the latest non major one). 

Looks like jupyter changed their API again, so we can't just take the latest.

File: src/client/common/process/environmentActivationService.ts [hunk 4]
```diff
@@ -112,6 +112,7 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
         @inject(IMemento) @named(GLOBAL_MEMENTO) private readonly memento: Memento,
         @inject(CondaService) private readonly condaService: CondaService,
         @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject('number')
```
Comment (rchiodo) [inline src/client/common/process/environmentActivationService.ts:115]: Inversify requires inject now even on optional items.

**Comment by rchiodo** [comment]
@DonJayamanne should be working now.

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**